app/components/two_column_layout/two_column_layout.js

```diff
@@ -0,0 +1,40 @@
+import { Controller } from 'stimulus';
+
+export default class extends Controller {
+  static targets = ['activeItem', 'sidebar'];
+
+  connect() {
+    if (this.hasActiveItemTarget && !this.isActiveItemVisible()) {
+      this.scrollActiveItemIntoView();
+    }
+  }
+
+  isActiveItemVisible() {
+    if (!this.hasActiveItemTarget) {
+      return;
+    }
+
+    const sidebar = this.sidebarTarget;
+    const sidebarBounds = sidebar.getBoundingClientRect();
+
+    const item = this.activeItemTarget;
+    const itemBounds = item.getBoundingClientRect();
+
+    // Calculate item bounds relative to sidebar
+    const itemTop = itemBounds.top - sidebarBounds.top - sidebar.scrollTop;
+    const itemBottom = itemTop + item.offsetHeight;
+
+    // Check if item is at least partially visible
+    return itemTop < sidebar.offsetHeight && itemBottom > 0;
+  }
+
+  scrollActiveItemIntoView() {
```

Comment:
This would be obsolete with fragment identifier.

Reply:
I thought linking to an anchor on the page might actually make this obsolete (even though it isn’t a solution to the code you commented on in [this comment](https://github.com/tactilenews/100eyes/pull/832/files/7f1c7204540d8ef864d936d1abd1bc361fe9c7c2..0c24ae9bbbf8e92de5764fd5f6688975719bf119#diff-11cfae26e2309368d917533b49425662d7d0206d3f7352e8746ac9116e7f618b). Unfortunately though, the browser isn’t smart enough to figure out the correct scroll position on its own… :(

<img width="1269" alt="Screenshot 2021-05-09 at 15 34 30" src="https://user-images.githubusercontent.com/1512805/117574217-5af17780-b0dc-11eb-8403-877edcdce811.png">